allensdk/internal/brain_observatory/time_sync.py

```diff
@@ -18,37 +18,51 @@
 LONG_STIM_THRESHOLD = 0.2     # seconds
 MAX_MONITOR_DELAY = 0.07     # seconds
 
-VERSION_1_KEYS = {
-    "photodiode": "stim_photodiode",
-    "2p": "2p_vsync",
-    "stimulus": "stim_vsync",
-    "eye_camera": "cam2_exposure",
-    "behavior_camera": "cam1_exposure",
-    "acquiring": "2p_acquiring",
-    "lick_sensor": "lick_1"
-    }
-
-# MPE is changing keys. This isn't versioned in the file.
-VERSION_2_KEYS = {
-    "photodiode": "photodiode",
-    "2p": "2p_vsync",
-    "stimulus": "stim_vsync",
-    "eye_camera": "eye_tracking",
-    "behavior_camera": "behavior_monitoring",
-    "acquiring": "2p_acquiring",
-    "lick_sensor": "lick_sensor"
-    }
-
-
-def get_keys(sync_dset):
-    """Get the correct lookup for line labels.
-
-    This method is fragile, but not all old data contains the full list
-    of keys.
+
+def get_keys(sync_dset: Dataset) -> dict:
+    """
+    Gets the correct keys for the sync file by searching the sync file
+    line labels. Assumes to use old keys if none of the correct ones are
+    found in line labels
+    Args:
+        sync_dset: The sync dataset to search for keys within
+
+    Returns:
+        key_dict: dictionary of key value pairs for finding data in the
+                  sync file
     """
-    if "cam2_exposure" in sync_dset.line_labels:
-        return VERSION_1_KEYS
-    return VERSION_2_KEYS
+    key_dict = {
+            "photodiode": ["stim_photodiode", "photodiode"],
+            "2p": ["2p_vsync"],
+            "stimulus": ["stim_vsync", "vsync_stim"],
+            "eye_camera": ["cam2_exposure", "eye_tracking",
+                           "eye_frame_received"],
+            "behavior_camera": ["cam1_exposure", "behavior_monitoring",
+                                "beh_frame_received"],
+            "acquiring": "2p_acquiring",
+            "lick_sensor": ["lick_1", "lick_sensor"]
+            }
+    label_set = set(sync_dset.line_labels)
+    remove_keys = []
+    for key, value in key_dict.items():
+        if isinstance(value, list):
+            value_set = set(value)
+            diff = value_set.intersection(label_set)
+            if len(diff) == 1:
+                key_dict[key] = diff.pop()
+            else:
+                logging.warning(f"Value not found for key {key}. Deleting"
+                                " key and value from dictionary")
+                remove_keys.append(key)
+    for key in remove_keys:
+        key_dict.pop(key)
+    line_label_set = set(sync_dset.line_labels)
+    non_found_keys = set(list(key_dict.values())) - line_label_set
+    if len(non_found_keys) > 0:
+        logging.warning("Keys not found in sync dataset line labels, assuming"
```

Comment:
How can a user take action on this warning message? It would be much more helpful to say which data source ('key') is missing lines.

Please change it so that it's like this:
> "Could not find valid lines for the following data sources:
 'photodiode' (valid line label(s) = ['photodiode', '2p_photodiode']),
'lick_sensor' (valid line label(s) = ['lick_1', 'lick_sensor'])"

etc.

Reply:
I moved to the suggested the logging output.

Reply:
what action was taken to resolve this conversation?

Reply:
see above, 'moved to suggested logging output'